torch_geometric/nn/models/sgformer.py

```diff
@@ -97,14 +105,17 @@ def forward(self, x):
         layer_.append(x)
 
         for i, attn in enumerate(self.attns):
-            x = attn(x)
+            x = attn(x, mask)
             x = (x + layer_[i]) / 2.
             x = self.bns[i + 1](x)
             x = self.activation(x)
             x = F.dropout(x, p=self.dropout, training=self.training)
             layer_.append(x)
 
-        return x
+        x_mask = x[mask]
+        # reverse the sorting
+        unsorted_x_mask = x_mask[indices.argsort()]
```

Comment:
technically can be also done in linear time :)

Reply:
i can do a follow up PR